3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py

```diff
@@ -63,46 +64,91 @@
 
 all_GSEA_results = []
 all_signatures = []
-#negative_control = []
-for col in signature.iloc[:,1:50].columns:
+neg_GSEA_results = []
+negative_control = []
+
+for col in signature.iloc[:,1:101].columns:
```

Comment:
try to avoid hardcoding. Is there a way to detect what the range to use is without harcoding?

Reply:
I changed the code and added a line to the following to extract the number of columns instead of hardcoding a certain range:
`range = signature.shape[1]`
`for col in signature.iloc[:,1:range].columns:`